### PR TITLE
Added new inflow from boundary condition options

### PR DESCRIPTION
**Pull request by IskanderI**
New feature: changing inflow boundary - this PR allows the inflow velocity to change through time.

I rewrote reading from `*.dat` file:
 1) now `bvel_velin` can be an array instead of a scalar and it writes to a different array. 
2) The time is given in a different array `velin_time_delims` which takes the corresponding velocity and assigns it as `bc.velin`.  

The same function calculates `bc.velout` as before. It is all similar to function `BCGetBGStrainRates`. 
I also added separate folder in tests (`t32_BC_velocity`) with an example of how to use it. This function doesn't change any workflow in usual models with only one velocity.
To use it:
-  specify _**velin_num_periods**_ (INT) for a numbers of changing periods
- _**velin_time_delims**_ (ARRAY) in Ma for an age when to change for a next period (one less than period number!)
- _**bvel_velin**_ (ARRAY) velocities for specified period

- With changing _**face_out**_ parameter to -1, 0 or 1 it is possible to change the behaviour of opposite boundary 
  - "-1"  can move the same way (e.g. compress from both sides or pull from both sides),
  - "0"  don 't move opposite direction (as before) 
  - "1" move the same direction (as before)

_lamem_input_ file has been updated accordingly

### CODE REVIEW

**Comment by IskanderI** [comment]
Should I do new PR with new description and close this?

**Comment by boriskaus** [comment]
no, this is fine. I initially didn't see your description, but I now reformatted this a little bit to make it more readable.

I have assigned Anton as reviewer.

**Comment by sgeotech** [comment]
I promise to review it promptly, within next three months or so :)

File: src/bc.cpp [hunk 1]
```diff
@@ -1457,6 +1464,32 @@ PetscErrorCode BCApplyVelDefault(BCCtx *bc)
...
+	PetscErrorCode ierr;
+	PetscFunctionBegin;
+	// initialize
+	time = bc->ts->time;
+	scal = bc->scal;
+	if(bc->VelNumPeriods)
+	{
+		for(jj = 0; jj < bc->VelNumPeriods-1; jj++)
+		{
+			if(time*scal->time < bc->VelTimeDelims[jj]*scal->time) break;
```
[the remark this answers is not in the document]
Comment (IskanderI) [reply src/bc.cpp]: I agree it is redundant here I removed it next commit see below

File: src/bc.cpp [hunk 2]
```diff
@@ -518,6 +518,14 @@ PetscErrorCode BCCreate(BCCtx *bc, FB *fb)
         ierr = getScalarParam(fb, _REQUIRED_, "bvel_top",    &bc->top,    1, scal->length  ); CHKERRQ(ierr);
         ierr = getScalarParam(fb, _REQUIRED_, "bvel_velin",  &bc->velin,  1, scal->velocity); CHKERRQ(ierr);
         ierr = getScalarParam(fb, _OPTIONAL_, "bvel_velout", &bc->velout, 1, scal->velocity); CHKERRQ(ierr);
+		ierr = getIntParam   (fb, _OPTIONAL_, "velin_num_periods",  &bc->VelNumPeriods,  1,                  _max_periods_  ); CHKERRQ(ierr);
+		ierr = getScalarParam(fb, _OPTIONAL_, "bvel_relax_d",&bc->relax_dist,1, scal->length  ); CHKERRQ(ierr);
+		if(bc->VelNumPeriods>1)
```
[the remark this answers is not in the document]
Comment (IskanderI) [reply src/bc.cpp:523]: If bc->VelNumPeriods is equal to 1 then it just takes 1 value for  bc->velin as in original code.

File: src/bc.cpp [hunk 3]
```diff
@@ -1748,12 +1784,24 @@ PetscErrorCode BCApplyBoundVel(BCCtx *bc)
             vel = 0.0;
             if(bc->face_out)
```
[the remark this answers is not in the document]
Comment (IskanderI) [reply src/bc.cpp:1785]: For some models it is necessary to push or pull from both sides of model. 
I dont know about pushing in one direction on both sides ( like plume movement modeling?) but I left it as in original imprementation.

**Pull request merged**